Assets/Scripts/Combat/Action.cs

```diff
@@ -78,6 +78,7 @@ virtual public void BeginAction(Tile targetTile)
         spentActionPoints = 0;
         inProgress = true;
         cooldownProgress = COOLDOWN;
+        if (DisplayName() != "") mechanics.DisplayPopup(DisplayName());
```

Comment:
I would incorporate the check for the null or empty string inside the CreatePopupText.
It's not the Action's or the Mechanics' job to check if the string is empty.

Reply:
That makes sense - done.